crates/red_knot_python_semantic/resources/mdtest/comprehensions/basic.md

```diff
@@ -43,7 +43,7 @@ class IntIterable:
     def __iter__(self) -> IntIterator:
         return IntIterator()
 
-# revealed: tuple[int, int]
+# revealed: tuple[int, Unknown | int]
```

Comment:
every comprehension has its own scope, so we also look up `y` as a public symbol :face_with_diagonal_mouth:

Reply:
I believe this is something @AlexWaygood wanted to look into, if I'm not mistaken (after outcome)

Reply:
Yes, for nested scopes that are known to execute eagerly (list/set/dict comprehensions, class bodies), and for ones that probably execute eagerly (generator expressions), we should model the name lookup as a "use" in the outer scope, at that point in control flow of the outer scope, rather than as a public type lookup. Alex has WIP on this, and it should fix this TODO and the one below.